docs/source/tutorial.rst

```diff
@@ -0,0 +1,511 @@
+  tutorial above and your objective to determine which files are needed for your
+  node.
+* Both files are enclosed under `namespace`. Look at other header-source file
+  pairs in ``robocup-software/soccer/src`` and determine what affects the
+  namespace. Set this up for your SoccerMom node under a folder called tutorial.
+* The header file includes important information regarding class structure and
+  inheritance (inheritance is important since the each node inherits from the
+  rclcpp::Node class). Moreover, it has a call to the source constructor in the
+  public access space.
+* Explore all the aspects of these files that applies to your task in this
+  section. Compare the similarities and differences between the subscribers and
+  publishers in these files vs. the ROS tutorial.
+* You will notice many unique operations and expressions in our codebase that
+  you might not have experience with. For instance in the radio.cpp file, find
+  the following expression:
+
+.. code-block::
+
+   create_subscription<rj_msgs::msg::ManipulatorSetpoint>(
+            control::topics::manipulator_setpoint_pub(i), rclcpp::QoS(1), [this,
+            i](rj_msgs::msg::ManipulatorSetpoint::SharedPtr manipulator) {  //
+            NOLINT
+                manipulators_cached_.at(i) = *manipulator;
+            });
+
+A ``lambda expression`` is used in place of the topic callback function in the
+ROS tutorial. The lambda expression allows you to pass in-line function objects
+and requires less lines of code when compared to having another function. I
+HIGHLY recommend that you google and visit the following resources to learn more
+about this topic.
+
+    - `<https://www.programiz.com/cpp-programming/lambda-expression>`_
+    - `<https://riptutorial.com/cplusplus/example/1854/what-is-a-lambda-expression->`_
+
+Another topic you may not be accustomed to is the arrow operation "->". For
+example:
+
+.. code-block::
+
+   robot_status_pubs_.at(robot_id)->publish(robot_status);
+
+The arrow operator is used to access a method or element of an object, when
+given a pointer to that object. Above, ``robot_status_pubs_`` is a list of
+pointers to ROS publisher objects. Calling `->publish(robot_status)` on one
+element in that list publishes a robot status using that specific publisher.
+
+    - `<https://www.tutorialspoint.com/cplusplus/cpp_member_operators.htm#:~:text=The%20(%2D%3E)%20arrow%20operator&text=The%20%2D%3E%20is%20called%20the%20arrow,pointer%2C%20use%20the%20arrow%20operator>`
+
+* Notice the docstrings for the radio header file. It explains that it's the
+  abstract superclass of the network_radio and sim_radio node. If you are
+  unfamiliar with the concept of abstraction, `here
+  <https://www.pythontutorial.net/python-oop/python-abstract-class/>`_ is more
+  information. When exploring these two nodes, you will discover a third file
+  named "node"_main.cpp which contains the main function for its respective
```

Comment:
```suggestion
  named <node>_main.cpp which contains the main function for its respective
```
this is the standard way to show that the text given is not literal, just a filler